libs/external-router/src/router/helper/insert-route-before-wildcard.ts

```diff
@@ -0,0 +1,21 @@
+import { Route, Routes } from '@angular/router';
+import { DaffExternalRouterNoWildcardError } from '../../errors/no-wildcard';
+
+/**
+ * Inserts a Route into Routes right before the wildcard route.
+ * If no wildcard is found, it fails fast with a `DaffExternalRouterNoWildcardError`.
+ */
+export const insertRouteBeforeWildCard = (
+	route: Route,
+	routes: Routes,
+): Routes => {
+	const index = routes.findIndex((r: Route) => r.path === '**');
+	if (index === -1) {
+		throw new DaffExternalRouterNoWildcardError(
+			'No wildcard (**) route was found during route resolution.',
+		);
+	}
+
+	routes.splice(index, 0, route);
+	return [...routes];
```

Comment:
Splice returns the deleted elements https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/splice

Reply:
I didn't say to return `routes.splice(index, 0, route)`. I'm saying, can't you just return `routes` instead of `[...routes]`. Like this:

```
routes.splice(index, 0, route);
return routes;
```

Reply:
Hrm Router. `config`, where this ends up, isn't explicitly required to be immutable, but I generally prefer immutable functions. It does bring up the point of where immutability should occur in a codebase, and maybe this isn't quite the right place.

Reply:
I could see the argument for immutability. But it seemed odd to change the routes with the splice, but then not to return it. I think you should go full immutable or not at all. What I mean is to do something like:
```
return [
   ...routes.split(0, index),
   route,
   ...routes.split(index+1, routes.length
];
```
Something like that would make it so the `routes` don't change.

Reply:
be4a79e3